File: typescript-client-sdk/src/sdk/models/shared/codatdatacontractsdatasetscommercepaymentmethodpagedresponselinksmodel.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { CodatDataContractsDatasetsCommercePaymentMethodPagedResponseHrefModel } from "./codatdatacontractsdatasetscommercepaymentmethodpagedresponsehrefmodel";
import { Expose, Type } from "class-transformer";


export class CodatDataContractsDatasetsCommercePaymentMethodPagedResponseLinksModel extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "current" })
  @Type(() => CodatDataContractsDatasetsCommercePaymentMethodPagedResponseHrefModel)
  current?: CodatDataContractsDatasetsCommercePaymentMethodPagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "next" })
  @Type(() => CodatDataContractsDatasetsCommercePaymentMethodPagedResponseHrefModel)
  next?: CodatDataContractsDatasetsCommercePaymentMethodPagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "previous" })
  @Type(() => CodatDataContractsDatasetsCommercePaymentMethodPagedResponseHrefModel)
  previous?: CodatDataContractsDatasetsCommercePaymentMethodPagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "self" })
  @Type(() => CodatDataContractsDatasetsCommercePaymentMethodPagedResponseHrefModel)
  self?: CodatDataContractsDatasetsCommercePaymentMethodPagedResponseHrefModel;
}

File: python-client-sdk/src/sdk/models/shared/codat_datacontracts_datasets_commerce_paymentmethodpagedresponselinksmodel.py
from __future__ import annotations
import dataclasses
from ..shared import codat_datacontracts_datasets_commerce_paymentmethodpagedresponsehrefmodel as shared_codat_datacontracts_datasets_commerce_paymentmethodpagedresponsehrefmodel
from dataclasses_json import Undefined, dataclass_json
from sdk import utils
from typing import Optional


@dataclass_json(undefined=Undefined.EXCLUDE)
@dataclasses.dataclass
class CodatDataContractsDatasetsCommercePaymentMethodPagedResponseLinksModel:
    current: Optional[shared_codat_datacontracts_datasets_commerce_paymentmethodpagedresponsehrefmodel.CodatDataContractsDatasetsCommercePaymentMethodPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('current'), 'exclude': lambda f: f is None }})
    next: Optional[shared_codat_datacontracts_datasets_commerce_paymentmethodpagedresponsehrefmodel.CodatDataContractsDatasetsCommercePaymentMethodPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('next'), 'exclude': lambda f: f is None }})
    previous: Optional[shared_codat_datacontracts_datasets_commerce_paymentmethodpagedresponsehrefmodel.CodatDataContractsDatasetsCommercePaymentMethodPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('previous'), 'exclude': lambda f: f is None }})
    self: Optional[shared_codat_datacontracts_datasets_commerce_paymentmethodpagedresponsehrefmodel.CodatDataContractsDatasetsCommercePaymentMethodPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('self'), 'exclude': lambda f: f is None }})
    